docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,197 @@
+  something metalK8s needs to provide as a feature. This can be achieved using
+  external log centralisation system.
+
+- stable ingestion: It is important to guaranty stable ingestion of the logs
+  and less important to guaranty stable performances when browsing/searching
+  the logs. However, peak loads related to complex logs queries should not
+  impact the application workloads and deploying log storage and search service
+  on infra nodes might help achieving this isolation.
+
+- stream indexing: It is not required to have automatic indexing of logs
+  content. Instead, the log centralisation service should offer basic features
+  to group/filter logs per tag/metadata defining the log stream.
+
+*Accessible from a central UI/API*
+
+Platform Admin or Storage Admin can visualize logs from all containers from all
+namespaces as well as journal logs, including (kubelet, containerd, salt-minion
+, kernel, initrd, services, etc ...) in Grafana. One can correlate logs and
+metrics or alerts in one single Grafana Dashboard.
+Browsing logs can be achieved through a documented API in order to expose logs
+in MetalK8s UIs or other workloads UIs if needed.
+
+*Persistence, Retention*
+
+Logs should be stored on a persistent storage. Platform Administrator can
+configure a max retention period. Some automatic purging mechanism is triggered
+when logs are older than the retention period or when the persistent storage is
+about to reach its capacity limit.
+A typical and default retention period is of 2 weeks.
+
+*Horizontally scalable (capacity and performance)*
+
+The Platform Administrator can add nodes or disks in order to increase storage
+capacity for logs (to keep more history or to store more workloads logs) or to
+increase performances for ingesting or querying the logs. It is also possible
+to fine tune the deployment configuration in order to have more replicas and
+ideally the Log centralisation service supports horizontal POD autoscaling.
+
+*Highly Available*
+
+Log collection, storage and browsing services can be replicated in order to
+ensure the same level of High Availability Prometheus has i.e. can lose all
+replicas but one.
+
+*Log Querying*
+
+Get all logs for a given period, node(s), pod regex, limited list of predefined
+labels and free keywords text.
+Typical Zenko use case: collecting all logs across several components, related
+to a S3 uniq request.
+
+*Log statistics (nice to have)*
+
+The Log centralisation service also offer the ability to consume statistics
+about the logs like the number of occurrences of one type of log during a
+certain period of time.
+
+*Monitorable/Observable service (health, performances and alerts)*
+
+The Platform Administrator can monitor capacity usage, IOPS, latency and
+bandwidth of the Log centralisation service. He can also monitor the health of
+the service (i.e. if some active alerts exist). He can visualize active and
```

Comment:
Can we define the scope of those alerts beforehand? E.g. monitoring the remaining capacity may simply be handled at the Volume level.

Reply:
I gave 2 examples of things that could impact the log service but at the end what the platform admin wants is to be notified when the service is degraded. Root cause is part of the alert notification.